packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -67,6 +69,34 @@ const ProductDetail = () => {
     const toast = useToast()
     const navigate = useNavigation()
     const customerId = useCustomerId()
+    const {site} = useMultiSite()
+    const storeInfoKey = `store_${site.id}`
+
+    // --- Add state for inventoryId ---
+    const [selectedInventoryId, setSelectedInventoryId] = useState(() => {
+        try {
+            return JSON.parse(window.localStorage.getItem(storeInfoKey))?.inventoryId || null
+        } catch (e) {
+            return null
+        }
+    })
```

Comment:
This is not a good React practise to listen to a value change.

If you want to create listener for any value change, you should useEffect and put your logic there. 
Using useState is not a way to listen for a change. useState is a way to trigger a re-rendering by changing the state.

Reply:
Sure, thanks I will refactor it